EDKII-DRAFT-RFCs/edkii-crypto-layout.md

```diff
@@ -0,0 +1,561 @@
+# RFC: 0000: EDK2 Crypto Architecture Reorganization
+
+## 1. Summary
+
+This proposal reorganizes EDK2's cryptographic components to improve modularity, maintainability, and security response time. The core changes are:
+
+1. **CryptoPkg remains in EDK2** as the standardized cryptographic interface layer (BaseCryptLib.h, TlsLib.h, HashApiLib.h)
+2. **Crypto provider implementations move to edk2-crypto repository** - OpensslPkg and MbedTlsPkg are promoted to independent packages
+3. **BaseCryptLib.h becomes the enforced contract** that all crypto providers must implement
+
+---
+
+## 2. Motivation
+
+The current monolithic CryptoPkg architecture creates several challenges that this proposal aims to address:
+
+### Independent Security Response
+
+Critical OpenSSL/MbedTLS vulnerabilities currently require full EDK2 release cycles, delaying security patches by weeks or months. By separating crypto providers into independent packages, security fixes can be released immediately without waiting for EDK2 coordination.
+
+### Unified Crypto History
+
+Crypto-related changes are currently scattered across EDK2's history, mixed with unrelated platform commits. Separating crypto into its own repository creates focused git history, making security audits easier and tracking crypto evolution clearer.
+
+### Enforced Architectural Boundaries
```

Comment:
Somewhere around here I think you should also address the use case where third-party code is wrapped (e.g. the TPM reference code) that has a direct crypto provider dependency. In that case, this allows that feature (TPM reference code) to directly use the crypto provider through a clean interface without it being publicly exported from CryptoPkg.